tests/configuration.py

```diff
@@ -0,0 +1,280 @@
+        """
+        conf = self._build_test_config()
+
+        parser = argparse.ArgumentParser()
+        parser.add_argument('--newkey')
+        parser.add_argument('--testkey')
+        parser.add_argument('--authorized_users')
+        parser.add_argument('--plugin-testplugin-testkey')
+        ns = parser.parse_args(['--testkey', 'testvalue'])
+
+        update_dict = {
+            'newkey': 'newvalue',
+            'authorized_users': ['user1'],
+            'plugin-testplugin-testkey': 'plugin-value',
+        }
+
+        f = io.StringIO()
+        with contextlib.redirect_stdout(f):
+            result = vars(conf.update(ns, update_dict))
+
+        self.assertTrue("--no-defaults" in f.getvalue())
+        self.assertEqual(result.get("newkey"), "newvalue")
+        self.assertEqual(result.get("testkey"), "testvalue")
+        self.assertTrue(isinstance(result.get("authorized_users"), list))
+        self.assertFalse(result.get("plugin-testplugin-testkey"))
+
+        f = io.StringIO()
+        sys.argv.append("--suppress-warnings")
+        with contextlib.redirect_stdout(f):
+            result = vars(conf.update(ns, None))
+        sys.argv.remove("--suppress-warnings")
+
+        self.assertFalse("--no-defaults" in f.getvalue())
+
+    def test_write_config(self):
+        """
+        Test CLIConfig.write_config()
+        """
+        conf = self._build_test_config()
+
+        conf.config.set("cli-dev", "type", "newvalue")
+        m = mock_open()
+        with patch('builtins.open', m):
+            conf.write_config()
+        self.assertIn(call("type = newvalue\n"), m().write.call_args_list)
+
+    def test_configure_no_default_terminal(self):
+        """
+        Test CLIConfig.configure() with
+        no default user, no environment variables, and no browser
+        """
+        conf = configuration.CLIConfig(self.base_url, skip_config=True)
+
+        def mock_input(prompt):
+            answers = (a for a in ["1", "1", "1", "1"])
+            if 'token' in prompt.lower():
+                return "test-token"
+            return next(answers)
+
+        with (patch('linodecli.configuration.open', mock_open()),
+                patch('builtins.input', mock_input),
+                contextlib.redirect_stdout(io.StringIO()),
+                patch.dict(os.environ, {}), requests_mock.Mocker() as m):
+            m.get(f'{self.base_url}/profile', json= {"username": "cli-dev"})
+            m.get(f'{self.base_url}/profile/grants', status_code=204)
+            m.get(f'{self.base_url}/regions', json= {"data":[{"id": "test-region"}]})
+            m.get(f'{self.base_url}/linode/types', json= {"data":[{"id": "test-type"}]})
+            m.get(f'{self.base_url}/images', json= {"data":[{"id": "test-image"}]})
+            m.get(f'{self.base_url}/account/users',
+                  json= {"data":[{"username": "cli-dev", "ssh_keys": "testkey"}]})
+            conf.configure()
```

Comment:
Getting this error here; thinking we just need an additional mock statement:
```
requests_mock.exceptions.NoMockAddress: No mock address: POST https://linode-test.com/profile/tokens
```

Reply:
weird it shouldn't be calling that url, i guess i need to make sure its not actually going to go to browser

Reply:
For reference it's being called from here: https://github.com/linode/linode-cli/blob/8e7c69f174a34c9f132ea7fcfe606cd2ed637c3c/linodecli/configuration/auth.py#L124

Wonder why it's using web auth :thinking:

Reply:
i added a mock to force it to false that should fix it

Reply:
Looks like your patch wasn't quite working because the current patch references `_check_browsers`  in the `helpers` package rather than the `configuration` package where it is imported. This is because it effectively copies the function ref when using `from ... import ...`. Just a bit of Python jankiness lol

This should fix it up :slightly_smiling_face: 

```python
patch('linodecli.configuration._check_browsers', lambda: False),
```

Reply:
updated good catch